metadata-io/src/test/java/com/linkedin/metadata/search/SearchServiceTest.java

```diff
@@ -160,4 +167,203 @@ public void testSearchService() throws Exception {
     searchResult = _searchService.searchAcrossEntities(ImmutableList.of(), "test", null, null, 0, 10, null);
     assertEquals(searchResult.getNumEntities().intValue(), 0);
   }
+
+  @Test
+  public void testAdvancedSearchOr() throws Exception {
+    final Criterion filterCriterion =  new Criterion()
+        .setField("platform")
+        .setCondition(Condition.EQUAL)
+        .setValue("hive")
+        .setValues(new StringArray(ImmutableList.of("hive")));
+
+    final Criterion subtypeCriterion =  new Criterion()
+        .setField("subtypes")
+        .setCondition(Condition.EQUAL)
+        .setValue("")
+        .setValues(new StringArray(ImmutableList.of("view")));
+
+    final Filter filterWithCondition = new Filter().setOr(
+        new ConjunctiveCriterionArray(
+            new ConjunctiveCriterion().setAnd(
+                new CriterionArray(ImmutableList.of(filterCriterion))),
+            new ConjunctiveCriterion().setAnd(
+                new CriterionArray(ImmutableList.of(subtypeCriterion)))
+        ));
+
+
+    SearchResult searchResult =
+        _searchService.searchAcrossEntities(ImmutableList.of(ENTITY_NAME), "test", filterWithCondition, null, 0, 10, null);
+
+    assertEquals(searchResult.getNumEntities().intValue(), 0);
+    clearCache();
+
+    Urn urn = new TestEntityUrn("test", "testUrn", "VALUE_1");
+    ObjectNode document = JsonNodeFactory.instance.objectNode();
+    document.set("urn", JsonNodeFactory.instance.textNode(urn.toString()));
```

Comment:
Nit: can we pull out some of these constants that we reuse? Also keeps things more extensible if we add more tests